src/replication_core.rs

```diff
@@ -20,57 +20,88 @@ impl Plugin for ReplicationCorePlugin {
     }
 }
 
+#[derive(Clone, Copy)]
+pub enum SendPolicy {
+    /// Unreliable and Unordered
+    Unreliable,
+    /// Reliable and Unordered
+    Unordered,
+    /// Reliable and Ordered
+    Ordered,
+}
+
 /// A resource to create channels for [`bevy_renet::renet::ConnectionConfig`]
 /// based on number of added server and client events.
-#[derive(Clone, Copy, Default, Resource)]
+#[derive(Clone, Resource)]
 pub struct NetworkChannels {
-    /// Increments with each server event registration.
-    server: u8,
-    /// Increments with each client event registration.
-    client: u8,
+    /// Grows with each server event registration.
+    server: Vec<SendPolicy>,
+    /// Grows with each client event registration.
+    client: Vec<SendPolicy>,
+}
+
+impl Default for NetworkChannels {
+    fn default() -> NetworkChannels {
+        NetworkChannels {
+            server: Vec::with_capacity(u8::MAX as usize),
+            client: Vec::with_capacity(u8::MAX as usize),
+        }
+    }
 }
 
 impl NetworkChannels {
     pub fn server_channels(&self) -> Vec<ChannelConfig> {
-        channel_configs(self.server)
+        channel_configs(&self.server)
     }
 
     pub fn client_channels(&self) -> Vec<ChannelConfig> {
-        channel_configs(self.client)
+        channel_configs(&self.client)
     }
 
-    pub(super) fn create_client_channel(&mut self) -> u8 {
-        self.client += 1;
-        self.client + REPLICATION_CHANNEL_ID
+    pub(super) fn create_client_channel(&mut self, send_policy: SendPolicy) -> u8 {
+        if self.client.len() == (REPLICATION_CHANNEL_ID + u8::MAX).into() {
+            panic!("Max client channels exceeded.");
```

Comment:
Panic messages should start with non-capital letter:
```suggestion
            panic!("max client channels exceeded u8::MAX.");
```